src/commands/node.mjs

```diff
@@ -664,45 +680,48 @@ export class NodeCommand extends BaseCommand {
       throw new FullstackTestingError(`Error starting node: ${e.message}`, e)
     } finally {
       await self.accountManager.close()
+      await self.close()
     }
 
     return true
   }
 
-  async checkNetworkNodeProxyUp (namespace, nodeId, maxAttempts = 10, delay = 5000) {
-    const podArray = await this.k8.getPodsByLabel([`app=haproxy-${nodeId}`, 'fullstack.hedera.com/type=haproxy'])
+  /**
+   * Check if the network node proxy is up, requires close() to be called after
+   * @param namespace the namespace
+   * @param nodeId the node id
+   * @param localPort the local port to forward to
+   * @param maxAttempts the maximum number of attempts
+   * @param delay the delay between attempts
+   * @returns {Promise<boolean>} true if the proxy is up
+   */
+  async checkNetworkNodeProxyUp (namespace, nodeId, localPort, maxAttempts = 10, delay = 5000) {
```

Comment:
`namespace` is already in configManager and k8 has access to it. so no need to really pass it to k8.

Reply:
I needed it for overriding in the test case when I was testing locally and had it pointed to a different cluster.

Reply:
Also, stateless/functional programming is more scalable in the long run.  The current way provides more flexibility when testing and future usage.

Reply:
In the test, we could just call `configManager.setFlag(flags.namespace, <namespace>)` to set new namespace. This is how other tests are testing different namespace scenario: see an example [here](https://github.com/hashgraph/solo/pull/127/files#diff-f5f3a03bb79f5baee73a5e182b7a8bd32b8c9bf587d5f52f9fc294da986e73dcR79). Let me know if that does not solve your test scenario.

Reply:
If we pass `namespace` in those two methods of K8, we should update the API for all methods in K8 for consistency.

Reply:
updated